src/services/factories/tokenList.ts

```diff
@@ -130,19 +132,40 @@ export function getTokensFactory(
     return new Map(tokenAddressIdPairs)
   }
 
+  async function fetchTcrAddresses(networkId: number): Promise<Set<string>> {
+    return new Set(await tcrApi.getTokens(networkId))
+  }
+
   async function updateTokenDetails(networkId: number, numTokens: number): Promise<void> {
-    // Fetch addresses from contract given numTokens count
-    const addressesAndIds = await retry(() => fetchAddressesAndIds(networkId, numTokens))
+    const [tcrAddressesSet, addressesAndIds] = await Promise.all(
+      // Double await yes! `retry` returns a Promise<Promise<>>
+      await Promise.all([
```

Comment:
This is such a hack, :rofl:

Reply:
Well, no one was there to stop me, and it worked...

:shrug: